### DISCUSSION TITLE
Gas spend

### DISCUSSION THREAD

**Discussion by getairdrop24** [zkSync-Community-Hub/zksync-developers#356] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

I am particularly interested in obtaining a breakdown of gas costs related to various transactions, smart contract interactions, and any other activities that have incurred gas fees on the ZKSync Era network. Additionally, if there are any tools or resources available for users to monitor their gas usage more efficiently, I would appreciate any guidance or information on accessing such features.

### What is the specific issue or error you're encountering?

![Screenshot 2024-02-20 193154](https://github.com/zkSync-Community-Hub/zksync-developers/assets/85836509/0698b492-bac1-4f28-9172-023f3ce2d65a)

I hope my concern is clear. Similar to the Ethereum chain, where users can view their spent fees, ZKSync Era should also incorporate this feature. This functionality would assist users in easily tracking their spent fees on the chain.

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

N/A

**Comment by AlbionaHoti** [comment]
Hi, @getairdrop24, thanks for reaching out! 

I will share some answers we've given users on GitHub Discussions regarding the gas fees on zkSync Era.

1. https://github.com/zkSync-Community-Hub/zksync-developers/discussions/203#discussioncomment-7774679
2. https://github.com/zkSync-Community-Hub/zksync-developers/discussions/287#discussioncomment-8318963

For external platforms to monitor gas usage more efficiently, I'll get back if I get a response from a colleague.

**Reply by AlbionaHoti** [reply]
Dune.com is a platform you can query these data from; it looks like Etherscan doesn't have these stats yet.

**Reply by getairdrop24** [reply]
The third-party services are not functioning properly, necessitating official support. Additionally, these third-party services are not deemed trustworthy.

**Reply by Malachy999** [reply]
any idea if this will be incorporated into Etherscan in future?

**Comment by jablonskibrittany** [comment]
Gas cost breakdown and monitoring tools are essential for users to track their spending efficiently on the ZKSync Era network. Implementing a feature that provides a breakdown of gas costs related to transactions, smart contract interactions, and other activities on the network would greatly enhance user experience. Additionally, providing tools or resources for users to monitor their gas usage would fur

**Comment by berronesjosh** [comment]
Implementing a gas cost breakdown feature in ZKSync Era would enhance user transparency and facilitate efficient monitoring of gas usage on the network.

**Comment by renemilnev** [comment]
Implementing a gas usage breakdown feature on zkSync Era would enhance transparency and help users track their spending more efficiently. This functionality would greatly benefit both users and developers by providing insights into transaction costs and optimizing gas usage on the network.

**Comment by dutterbutter** [comment, marked as answer]
@getairdrop24 please make use of era-test-node's `--show-gas-details` feature for full tx breakdown.

**Comment by queskim** [comment]
To obtain a breakdown of gas costs related to various transactions and smart contract interactions on the ZKSync Era network, we can implement a feature that allows users to view their spent fees. Additionally, we can provide tools and resources to help users monitor their gas usage more efficiently.
